lib/chartjs/chart_helpers.rb

```diff
@@ -49,16 +53,9 @@ def chart(klass, data, options)
 
             #{legend if generate_legend}
           };
-          /* W3C standard */
-          if (window.addEventListener) {
-            window.addEventListener("load", initChart, false);
-            document.addEventListener("page:load", initChart, false);
-          }
-          /* IE */
-          else if (window.attachEvent) {
-            window.attachEvent("onload", initChart);
-            document.attachEvent("page:load", initChart);
-          }
+          
+          initChart();
```

Comment:
Presumably your method works because by the time all this stuff is rendered the `onload` and `page:load` events have already fired?

I don't understand why the existing implementation doesn't work with ajax.  Please could you elaborate on the problem?